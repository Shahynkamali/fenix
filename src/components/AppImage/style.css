.image {
  @apply inline-block relative;
}

.image__placeholder {
  @apply absolute overflow-hidden;
}

.image__placeholder img {
  @apply scale-105;

  filter: blur(10px);
}

.image__img {
  @apply opacity-0;

  transition: opacity 300ms ease;
}

.image--loaded .image__img {
  @apply opacity-100;
}
