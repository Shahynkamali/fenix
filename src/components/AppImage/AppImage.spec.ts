import { mount } from '@vue/test-utils';
import setupIntersectionObserverMock from '@/tests/__mocks__/intersection';
import AppImage from './AppImage.vue';

describe('AppImage', () => {
  const factory = (propsData?: any) => mount(AppImage, {
    propsData: {
      ...propsData,
    },
  });

  beforeEach(() => {
    setupIntersectionObserverMock();
  });

  it('should render a image', () => {
    const srcProp = {
      src: 'https://picsum.photos/200',
    };
    const wrapper = factory(srcProp);
    expect(wrapper.vm.$props.src).toBe(srcProp.src);
    expect(wrapper.find('.image__img')).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render a placeholder component', () => {
    const srcProp = {
      src: 'https://picsum.photos/200',
      placeholder: 'https://picsum.photos/200/?blur',
    };
    const wrapper = factory(srcProp);
    expect(wrapper.find('.image__placeholder')).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
