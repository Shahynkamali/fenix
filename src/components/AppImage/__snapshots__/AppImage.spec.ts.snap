// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppImage should render a image 1`] = `
<div
  class="image"
>
  <!---->
   
  <img
    alt="image"
    class="image__img"
    src=""
  />
</div>
`;

exports[`AppImage should render a placeholder component 1`] = `
<div
  class="image"
>
  <!---->
   
  <img
    alt="image"
    class="image__img"
    src=""
  />
</div>
`;
